projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -204,6 +205,31 @@ export class CodeViewerComponent implements AfterViewInit, OnChanges, OnDestroy
     });
   }
 
+  private isLineHighlighted(lineNum: number): boolean {
+    /**
+     * `isEmpty(new RexExp('anything'))` returns true, so here we need to make another check for RegExp
+     */
+    const emptyValue =
```

Comment:
can we keep out some checks outside this method? the highlightText is only changing once so we don't need to check its type for every line.

we could also avoid lowercase conversation multiple times for the same line.